### PR TITLE
Add E2E job which build and deploy operators

### PR DESCRIPTION
**Pull request by Sandeepyadav93**
This patch adds a job to build and deploy neutron-operator.

### CODE REVIEW

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-neutron-operator-build-deploy

**Comment by Sandeepyadav93** [comment]
Pull was failing, updated image to public on quay
~~~
$ podman pull quay.rdoproject.org/openstack-k8s-operators/neutron-operator-bundle:05ad70ad63650460182c35650d2b16b5d267fdb4
Trying to pull quay.rdoproject.org/openstack-k8s-operators/neutron-operator-bundle:05ad70ad63650460182c35650d2b16b5d267fdb4...
Error: initializing source docker://quay.rdoproject.org/openstack-k8s-operators/neutron-operator-bundle:05ad70ad63650460182c35650d2b16b5d267fdb4: reading manifest 05ad70ad63650460182c35650d2b16b5d267fdb4 in quay.rdoproject.org/openstack-k8s-operators/neutron-operator-bundle: unauthorized: access to the requested resource is not authorized
~~~

File: ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml [hunk 1]
```diff
@@ -9,6 +9,26 @@ build_root:
     name: release
     namespace: openshift
     tag: rhel-8-release-golang-1.18-openshift-4.12
```
Comment (Sandeepyadav93) [inline ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml:11]: Missed to bump golang 1.19 image.

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/36052/rehearse-36052-pull-ci-openstack-k8s-operators-neutron-operator-master-neutron-operator-build-deploy/1621872625884073984#1:build-log.txt%3A448

go: go.mod file indicates go 1.19, but maximum version supported by tidy is 1.18

Comment (Sandeepyadav93) [reply ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml]: done

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-neutron-operator-build-deploy

**Comment by Sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-neutron-operator-build-deploy

**Comment by sandeepyadav93** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-neutron-operator-build-deploy

**Review by karelyatin — APPROVED** [review]

**Pull request merged**